### PR TITLE
RAT-333: fix --force option will change the file permissions

### PR DESCRIPTION
**Pull request by Claudenw**
Fix for RAT-333

Added test to cover the issue as well as a general test to show that the "force" command works.

### CODE REVIEW

**Comment by ottlinger** [comment]
@Claudenw interesting catch! Thanks for fixing it - would you mind adding a changelog entry within the 0.17-block?

```xml
      <action issue="RAT-333" type="fix" dev="claudenw">
        Fix if --force option is used executable bit is not set properly on newly created/license-augmented file is created.
      </action>

```

Please do not change the format of the file as the maven-changelog-plugin is quite "picky" :)

**Comment by Claudenw** [comment]
> @Claudenw interesting catch! Thanks for fixing it - would you mind adding a changelog entry within the 0.17-block?

I think I got it correctly done.  Please verify.

File: apache-rat-core/src/main/java/org/apache/rat/annotation/AbstractLicenseAppender.java [hunk 1]
```diff
@@ -239,27 +241,28 @@ public void append(File document) throws IOException {
...
 
         if (isForced) {
-            boolean deleted = document.delete();
-            if (!deleted) {
-                log.error("Could not delete original file to prepare renaming.");
-            }
-            boolean renamed = newDocument.renameTo(document.getAbsoluteFile());
-            if (!renamed) {
-                log.error("Failed to rename new file, original file remains unchanged.");
+            Path docPath = document.toPath();
```
[the remark this answers is not in the document]
Comment (ottlinger) [reply apache-rat-core/src/main/java/org/apache/rat/annotation/AbstractLicenseAppender.java:257]: Should we move thw isExecutable into the try as well? @Claudenw

[the remark this answers is not in the document]
Comment (Claudenw) [reply apache-rat-core/src/main/java/org/apache/rat/annotation/AbstractLicenseAppender.java:257]: good catch

File: apache-rat-core/src/test/java/org/apache/rat/annotation/TestLicenseAppender.java [hunk 2]
```diff
@@ -674,4 +678,70 @@ public void readFile(BufferedReader r) throws IOException {
                     }
                 });
     }
+    
+    @Test
```
[the remark this answers is not in the document]
Comment (Claudenw) [reply apache-rat-core/src/test/java/org/apache/rat/annotation/TestLicenseAppender.java:683]: I reformatted my changes to match what was there (I think).

**Comment by ottlinger** [comment]
Thanks for spotting this error and providing a fix.

**Pull request merged**